### PR TITLE
[source-asana] Upgrade to concurrent for non-incremental substreams / list based

### PR DESCRIPTION
**Pull request by brianjlai**
## What

Upgrades to the latest CDK version to uptake concurrency. This is also needed as part of pre-release testing the latest change to the CDK which will make substreams that are not incremental run in the concurrent framework

## Review guide

n/a

## User Impact

faster syncs

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

state message format is not changing at this moment

### CODE REVIEW

**Comment by brianjlai** [comment]
retesting regression tests with the latest candidate
https://github.com/airbytehq/airbyte/actions/runs/12130315383/job/33820422548

- record counts match as expected
- different record values expected due to changes in null values in a prior cdk
- one stream marked as `is_resumable` == False, it used to be `is_resumable` == True. Expected since we have one full refresh (non-substream) that is no longer using RFR

I think we're in a good place to merge this change in once we release the new CDK which i'll then update to use a non-RC candidate

File: airbyte-integrations/connectors/source-asana/pyproject.toml
Comment (brianjlai) [reply airbyte-integrations/connectors/source-asana/pyproject.toml:19]: that's a long list, but basically:

- latest python / pydantic / etc
- not emitting null fields
- other CDK changes in past versions
- upgrade to concurrency
- (latest candidate version) Moving the stream grouping and creation out of the source init and back into `discover()` / `read()` etc

I did this to retest my changes before merging this PR: https://github.com/airbytehq/airbyte-python-cdk/pull/130

**Review by maxi297 — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**
